lightning/src/ln/channel.rs

```diff
@@ -1642,6 +1661,202 @@ impl<SP: Deref> InitialRemoteCommitmentReceiver<SP> for InboundV1Channel<SP> whe
+				channel_id: self.context().channel_id(),
+				data: b"No interactive transaction negotiation in progress".to_vec()
+			}),
+		})
+	}
+
+	fn tx_remove_output(&mut self, msg: &msgs::TxRemoveOutput)-> InteractiveTxMessageSendResult {
+		InteractiveTxMessageSendResult(match self.interactive_tx_constructor_mut() {
+			Some(ref mut tx_constructor) => tx_constructor.handle_tx_remove_output(msg).map_err(
+				|reason| reason.into_tx_abort_msg(self.context().channel_id())),
+			None => Err(msgs::TxAbort {
+				channel_id: self.context().channel_id(),
+				data: b"No interactive transaction negotiation in progress".to_vec()
+			}),
+		})
+	}
+
+	fn tx_complete(&mut self, msg: &msgs::TxComplete) -> HandleTxCompleteResult {
+		let tx_constructor = match self.interactive_tx_constructor_mut() {
+			Some(ref mut tx_constructor) => tx_constructor,
+			None => {
+				let tx_abort = msgs::TxAbort {
+					channel_id: msg.channel_id,
+					data: b"No interactive transaction negotiation in progress".to_vec(),
+				};
+				return HandleTxCompleteResult(Err(tx_abort));
+			},
+		};
+
+		let tx_complete = match tx_constructor.handle_tx_complete(msg) {
+			Ok(tx_complete) => tx_complete,
+			Err(reason) => {
+				return HandleTxCompleteResult(Err(reason.into_tx_abort_msg(msg.channel_id)))
+			}
+		};
+
+		if let HandleTxCompleteValue::SendTxComplete(_, ref signing_session) = tx_complete {
+			self.context_mut().next_funding_txid = Some(signing_session.unsigned_tx.txid());
+		};
+
+		HandleTxCompleteResult(Ok(tx_complete))
+	}
+
+	fn funding_tx_constructed<L: Deref>(
+		&mut self, signing_session: &mut InteractiveTxSigningSession, logger: &L
+	) -> Result<(msgs::CommitmentSigned, Option<Event>), ChannelError>
+	where
+		L::Target: Logger
+	{
+		let our_funding_satoshis = self.dual_funding_context().our_funding_satoshis;
+		let context = self.context_mut();
+
+		let mut output_index = None;
+		let expected_spk = context.get_funding_redeemscript().to_p2wsh();
+		for (idx, outp) in signing_session.unsigned_tx.outputs().enumerate() {
+			if outp.script_pubkey() == &expected_spk && outp.value() == context.get_value_satoshis() {
+				if output_index.is_some() {
+					return Err(ChannelError::Close(
+						(
+							"Multiple outputs matched the expected script and value".to_owned(),
+							ClosureReason::HolderForceClosed { broadcasted_latest_txn: Some(false) },
+						)));
+				}
+				output_index = Some(idx as u16);
+			}
+		}
+		let outpoint = if let Some(output_index) = output_index {
+			OutPoint { txid: signing_session.unsigned_tx.txid(), index: output_index }
+		} else {
+			return Err(ChannelError::Close(
+				(
+					"No output matched the funding script_pubkey".to_owned(),
+					ClosureReason::HolderForceClosed { broadcasted_latest_txn: Some(false) },
+				)));
+		};
+		context.channel_transaction_parameters.funding_outpoint = Some(outpoint);
+		context.holder_signer.as_mut().provide_channel_parameters(&context.channel_transaction_parameters);
```

Comment:
These are overridden in `get_initial_commitment_signed`, which is called next. Do they need to be set here?

Reply:
Will remove the duplication in `get_initial_commitment_signed`